packages/compass-editor/src/autocompleter.ts

```diff
@@ -0,0 +1,150 @@
+import { gte } from 'semver';
+import {
+  ACCUMULATORS,
+  BSON_TYPES,
+  BSON_TYPE_ALIASES,
+  CONVERSION_OPERATORS,
+  EXPRESSION_OPERATORS,
+  JSON_SCHEMA,
+  QUERY_OPERATORS,
+  STAGE_OPERATORS,
+} from '@mongodb-js/mongodb-constants';
+
+const ALL_COMPLETIONS = [
+  ...ACCUMULATORS,
+  ...BSON_TYPES,
+  ...BSON_TYPE_ALIASES,
+  ...CONVERSION_OPERATORS,
+  ...EXPRESSION_OPERATORS,
+  ...JSON_SCHEMA,
+  ...QUERY_OPERATORS,
+  ...STAGE_OPERATORS,
+];
+
+type Meta =
+  | typeof ALL_COMPLETIONS[number]['meta']
+  | 'field:identifier'
+  | 'field:reference';
+
+/**
+ * Our completions are a mix of ace autocompleter types and some custom values
+ * added on top, this interface provides a type definition for all required
+ * properties that completer is using
+ * @internal
+ */
+export type Completion = {
+  value: string;
+  version: string;
+  meta: Meta;
+  description?: string;
+  snippet?: string;
+  score?: number;
+};
+
+function matchesMeta(filter: string[], meta: string) {
+  const metaParts = meta.split(':');
+  return filter.some((metaFilter) => {
+    const filterParts = metaFilter.split(':');
+    return (
+      filterParts.length === metaParts.length &&
+      filterParts.every((part, index) => {
+        return part === '*' || part === metaParts[index];
+      })
+    );
+  });
+}
+
+export function createCompletionFilter(
+  prefix: string,
+  serverVersion: string,
+  filterMeta?: string[]
+) {
+  const currentServerVersion =
+    serverVersion.match(/^(?<version>\d+?\.\d+?\.\d+?)/)?.groups?.version ??
+    serverVersion;
+  return ({ value, version: minServerVersion, meta }: Completion) => {
+    return (
+      value.toLowerCase().startsWith(prefix.toLowerCase()) &&
+      gte(currentServerVersion, minServerVersion) &&
+      (!filterMeta || matchesMeta(filterMeta, meta))
+    );
+  };
+}
+
+export type CompleteOptions = {
+  // Current server version (default is 999.999.999)
+  serverVersion?: string;
+  // Additional fields that are part of the document schema to add to
+  // autocomplete as identifiers and identifier references
+  fields?: string[];
+  // Filter completions by completion value type
+  meta?: (Meta | 'field:*' | 'accumulator:*' | 'expr:*')[];
+};
+
+export type CompletionResult = {
+  // Autocomplete value that prefix was matched on
+  value: string;
+  // Value type
+  meta?: string;
+  // Longer description
+  description?: string;
+  // String representing a possible snippet completion
+  snippet?: string;
+  // For ace compat
+  score: number;
+};
+
+function isValidIdentifier(identifier: string) {
+  // Quick check for common case first
+  if (/[.\s"'()[\];]/.test(identifier)) {
```

Comment:
```suggestion
  if (/[.\s"'()[\];={}]/.test(identifier)) {
```

Otherwise something like `a=1` would pass the test here, right?

Reply:
Hah, yes, great catch, thank you